docs/extending/developing-code/code-overview/boxes-forms-and-formats.rst

```diff
@@ -0,0 +1,24 @@
+.. index:: Boxes
+.. index:: Forms
+
+Boxes, Forms, and Formats
+=========================
+
+In the introduction to this section we gave the 3-step process converting the input you type into output produced:
+
+* the input string is parsed to ``FullForm``
+* The ``FullForm`` S-expression is evaluated giving another S-expression
+* The result is formatted to the kind of output desired. Formatting can cause additional evaluations to occurs
+
+Here, we are going to go in more detail over the last step which broadly is formatting.
+
+At the top-level the kind of output you get is dictated by the Form specified either explicitly or implicitly. See `Forms of Input and Output <https://reference.wolfram.com/language/tutorial/TextualInputAndOutput.html#12368>`_.
+
+Formatting a "Form" often causes some parts of the form to be "boxed" as the S-expression is traversed. You can think of boxing as akin to adding parenthesis to Mathematical expressions written in normal infix notation. The parenthesis service to mark explicitly groupings. Other than that it does not change the meaning.
```

Comment:
Actually, this is more than adding parentheses: is to transform an expression that represents a composition of functions acting over certain symbols, into an expression that gives the typographical representation of that expression. For example, the input
``a/b+c``
leads to the Expression
``Plus[Times[a, Power[b,-1]], c]``
Then, ``ToBoxes[%]`` produces the expression
``RowBox[{"a", +, FractionBox["b", "c"]}]``
This Expression now is a "composition" of `Boxes`. Mathics then process these boxes to produce a (Python) string by calling `boxes_to_*` methods of the class Expression and its derivatives. For certain special cases of atoms (`Graphics`,  `CompiledCode`, `Image`), instead of an expression `ToBoxes` produces a `BoxConstruct` object, that shares much of its interface with an `Expression`, but is an `InstanceableBuiltin`.

Reply:
Yes, I have always had a nagging feeling that what I wrote wasn't totally accurate but a reasonable and _approximate_ way to think about this. And at some point I was going to go back and revise something along those lines.

If you can think of a simple way to express this great. Or maybe there could be a footnote. 

But in general, one of the problems we have along documentation here is like other things, we should at least get started in noting some of the current behavior down.

Creating Forms is really just another pass at Evaluation. And after something is evaluated there are these no kinds of objects that can get created like Graphics Objects, or DateTime Objects, or Boxing objects.

Reply:
@rocky, I know: I just write this because it is very difficult to put this in a plain clear way. And is difficult because the way in which all of this is done is not completely right. So, I think that to be able to write this in a clearer way, the best is to make this actually works in the core in a more clear way.